packages/@aws-cdk/aws-lambda/lib/edge-function.ts

```diff
@@ -0,0 +1,262 @@
+    return this.lambda.addPermission(id, permission);
+  }
+  public addToRolePolicy(statement: iam.PolicyStatement): void {
+    return this.lambda.addToRolePolicy(statement);
+  }
+  public grantInvoke(identity: iam.IGrantable): iam.Grant {
+    return this.lambda.grantInvoke(identity);
+  }
+  public metric(metricName: string, props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return this.lambda.metric(metricName, { ...props, region: EdgeFunction.EDGE_REGION });
+  }
+  public metricDuration(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return this.lambda.metricDuration({ ...props, region: EdgeFunction.EDGE_REGION });
+  }
+  public metricErrors(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return this.lambda.metricErrors({ ...props, region: EdgeFunction.EDGE_REGION });
+  }
+  public metricInvocations(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return this.lambda.metricInvocations({ ...props, region: EdgeFunction.EDGE_REGION });
+  }
+  public metricThrottles(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return this.lambda.metricThrottles({ ...props, region: EdgeFunction.EDGE_REGION });
+  }
+  public addEventSource(source: IEventSource): void {
+    return this.lambda.addEventSource(source);
+  }
+  public configureAsyncInvoke(options: EventInvokeConfigOptions): void {
+    return this.lambda.configureAsyncInvoke(options);
+  }
+
+  /** Create a function in-region */
+  private createInRegionFunction(id: string, props: FunctionProps): FunctionConfig {
+    const role = props.role ?? defaultLambdaRole(this, id);
+    const edgeFunction = new Function(this, 'Fn', {
+      ...props,
+      role,
+    });
+
+    return { edgeFunction, edgeArn: edgeFunction.currentVersion.edgeArn, functionStack: this.stack };
+  }
+
+  /** Create a support stack and function in us-east-1, and a SSM reader in-region */
+  private createCrossRegionFunction(id: string, props: FunctionProps): FunctionConfig {
+    const parameterName = `EdgeFunctionArn${id}`;
+    const functionStack = this.edgeStack();
+    this.stack.addDependency(functionStack);
+
+    const edgeFunction = functionStack.addEdgeFunction(id, parameterName, props);
+    // This is used to determine when the function has changed, to refresh the ARN from the custom resource.
+    const edgeFunctionHash = calculateFunctionHash(edgeFunction);
+
+    const parameterArn = this.stack.formatArn({
+      service: 'ssm',
+      region: EdgeFunction.EDGE_REGION,
+      resource: 'parameter',
+      resourceName: parameterName,
+    });
+
+    const resourceType = 'Custom::CrossRegionStringParameterReader';
+    const serviceToken = CustomResourceProvider.getOrCreate(this, resourceType, {
+      codeDirectory: path.join(__dirname, 'edge-function'),
+      runtime: CustomResourceProviderRuntime.NODEJS_12,
+      policyStatements: [{
+        Effect: 'Allow',
+        Resource: parameterArn,
```

Comment:
Will this work with multiple Lambda@Edge constructs? How I see it the `serviceToken` will be reused only has permission to get the first parameter.

Reply:
Great catch, thanks for the spot!